webknossos/webknossos/geometry/vec_int.py

```diff
@@ -0,0 +1,187 @@
+import re
+from operator import add, floordiv, mod, mul, sub
+from typing import Any, Callable, Iterable, List, Optional, Tuple, Union, cast
+
+import numpy as np
+
+VALUE_ERROR = "VecInt can be instantiated with int values `VecInt(1,2,3,4) or with `VecIntLike` object `VecInt([1,2,3,4])."
+
+
+class VecInt(tuple):
+    def __new__(
+        cls,
+        vec: Union[int, "VecIntLike"],
+        *args: int,
+        y: Optional[int] = None,
+        z: Optional[int] = None,
+        **kwargs: int,
+    ) -> "VecInt":
+        """
+        Class to represent a ND vector. Inherits from tuple and provides useful
+        methods and operations on top. The vector has a minimal length of 3 and
+        it is assumed that the first three axes are `x`, `y` and `z`.
+
+        A small usage example:
+
+        ```python
+        from webknossos import VecInt
+
+        vector_1 = VecInt(1, y=2, z=3, t=4)
```

Comment:
I agree. It would be possible, but only if the first positional argument is named `x` instead of `vec`. Yet the position of x would still be fixed and that of the other axes would be variable. I think it's a bit unfortunate that the VecInt class can be passed both a VecInt as a parameter and a set of int values.